### PR TITLE
Add macOS and watchOS support to Package.swift

### PR DESCRIPTION
**Pull request by mackoj**
Hi,

Thank you for creating this library, upon which I rely daily, along with my library [EnvironmentVariationPreview](https://github.com/mackoj/EnvironmentVariationPreview), to enhance the testing coverage of my work.

To enable the usage of `swift package diagnose-api-breaking-changes`, I kindly request adding support for `macOS` in the `Package.swift` file. Additionally, if feasible, it would be great to include support for `watchOS` as well.

Thank you once again!

What does this PR modify:
- Add missing import for UIKit.
- Add platforms availability marker.

Now, `swift package diagnose-api-breaking-changes main` seems to build the branch version, but it can't build the `main` branch for macOS since it is not compatible or buildable.

This PR don't change any logic of `swiftui-preview-snapshots` it just add a few `@available` marker.

Best regards,

### CODE REVIEW

**Comment by mackoj** [comment]
Hi @jflan-dd,

What do you think about this PR ?

File: Sources/PreviewSnapshotsTesting/PreviewSnapshots+assertSnapshots.swift [hunk 1]
```diff
@@ -11,9 +11,11 @@
 // or implied. See the License for the specific language governing permissions and limitations under
 // the License.
 
+#if canImport(UIKit)
```
Comment (avm0418) [inline Sources/PreviewSnapshotsTesting/PreviewSnapshots+assertSnapshots.swift:14]: I really hope watchOS support gets approved. Should this if import just wrap the UIKit import and why is this needed?

Comment (mackoj) [reply Sources/PreviewSnapshotsTesting/PreviewSnapshots+assertSnapshots.swift:14]: Because UIImage is used and UIKit import is missing.

**Comment by jflan-dd** [comment]
Hey @mackoj!

Thanks for the contribution!

Sorry I missed the notification on this PR. I'll take a look at it today.

**Comment by jflan-dd** [comment]
The `swift-snapshot-testing` library that this library is built on top of doesn't currently support watchOS in their package manifest. If there's useful snapshotting that can be done on watchOS during tests we could try and get supported added to their library.
https://github.com/pointfreeco/swift-snapshot-testing/blob/dc46eeb3928a75390651fac6c1ef7f93ad59a73b/Package.swift#L4-L10

**Comment by mackoj** [comment]
So let's remove watchOS for the moment I'll update the PR tomorrow.

**Comment by jflan-dd** [comment]
@mackoj I think putting all of the `assertSnapshots` methods inside of a `#if canImport(UIKit)` is going to make the library tough to use on macOS.

Maybe we should update the various `assertSnapshots` methods to be generic over the format snapshotting format, and then provide default implementations when UIKit is available.

```swift
    public func assertSnapshots<Format>(
        as snapshotting: Snapshotting<AnyView, Format>,
        record recording: Bool = false,
        file: StaticString = #file,
        testName: String = #function,
        line: UInt = #line
    ) {
        // Unchanged...
    }
```

and 

```swift
#if canImport(UIKit)
import UIKit

extension PreviewSnapshots {
    /// Doc comment
    public func assertSnapshots(
        record recording: Bool = false,
        file: StaticString = #file,
        testName: String = #function,
        line: UInt = #line
    ) {
        assertSnapshots(as: .image, record: recording, file: file, testName: testName, line: line)
    }
}

// MARK: - PreviewSnapshots.assertSnapshots + modify

extension PreviewSnapshots {
    /// Doc comment
    public func assertSnapshots<Modified: View>(
        record recording: Bool = false,
        file: StaticString = #file,
        testName: String = #function,
        line: UInt = #line,
        modify: (AnyView) -> Modified
    ) {
        assertSnapshots(as: .image, record: recording, file: file, testName: testName, line: line, modify: modify)
    }
}
#endif
```

**Comment by mackoj** [comment]
This is a better approach for sure my goal was merely to make it compile on macOS but having it working macOS is even better ? do you want to do it ? should I close my PR?

**Comment by mackoj** [comment]
#3 Is better than this PR so I close it
